sdk/src/main/java/com/horizen/utils/BytesUtils.java

```diff
@@ -190,4 +195,54 @@ public static boolean contains(Collection<byte[]> collection, byte[] value) {
         }
         return false;
     }
+
+    private static int HorizenPublicKeyAddressPrefixLength = 2;
+    private static int HorizenPublicKeyAddressHashLength = 20;
+    private static int HorizenPublicKeyAddressChecksumLength = 4;
+
+    private static int HorizenPublicKeyAddressBase58CheckLength = 35;
+
+    private static byte[] PublicKeyMainNetPrefix = BytesUtils.fromHexString("2089"); // "zn"
+    private static byte[] PublicKeyMainNetPrefixOld = BytesUtils.fromHexString("1CB8"); // "t1"
+
+    private static byte[] PublicKeyTestNetPrefix = BytesUtils.fromHexString("2098"); // "zt"
+    private static byte[] PublicKeyTestNetPrefixOld = BytesUtils.fromHexString("1D25"); // "tm"
+
+    public static byte[] fromHorizenPublicKeyAddress(String address, NetworkParams params) {
+        if(address.length() != HorizenPublicKeyAddressBase58CheckLength)
+            throw new IllegalArgumentException(String.format("Incorrect Horizen public key address length $d", address.length()));
+
+        byte[] addressBytesWithChecksum = Base58.decode(address).get();
+        byte[] addressBytes = Arrays.copyOfRange(addressBytesWithChecksum, 0, HorizenPublicKeyAddressPrefixLength + HorizenPublicKeyAddressHashLength);
+
+
+        // Check version
+        byte[] prefix = Arrays.copyOfRange(addressBytes, 0, HorizenPublicKeyAddressPrefixLength);
+        if(params instanceof MainNetParams) {
```

Comment:
better add parameter in params of public key address formatinstead of use instance of

Reply:
let's keep as is.
Maybe will be different after solving the main net case in Json encoder.